packages/blockchain-extension/extension/commands/submitTransaction.ts

```diff
@@ -30,8 +30,58 @@ import { GlobalState, ExtensionData } from '../util/GlobalState';
 interface ITransactionData {
     transactionName: string;
     transactionLabel?: string;
-    arguments?: string[];
-    transientData?: any;
+    arguments: string[];
+    transientData: any;
+}
+
+function validateTxnDataObject(txnDataObject: ITransactionData): ITransactionData {
+    let args: Array<string>;
+    let transientData: { [key: string]: Buffer };
+    if (txnDataObject.arguments) {
+        args = txnDataObject.arguments.map((arg: any) => typeof arg !== 'string' ? JSON.stringify(arg) : arg);
+    } else {
+        args = [];
+    }
+
+    transientData = txnDataObject.transientData || {};
+    const keys: Array<string> = Array.from(Object.keys(transientData));
+    if (keys.length > 0) {
+        keys.forEach((key: string) => {
+            transientData[key] = Buffer.from(transientData[key]);
+        });
+    }
+
+    return { ...txnDataObject, arguments: args, transientData };
+}
+
+async function createArgs(transactionObject: any): Promise<Array<string>> {
+    let argsString: string;
+    if (transactionObject && transactionObject.args) {
+        if (Array.isArray(transactionObject.args)) {
+            return transactionObject.args;
+        }
+        argsString = transactionObject.args;
+    } else {
+        argsString = await UserInputUtil.showInputBox('optional: What are the arguments to the transaction, (e.g. ["arg1", "arg2"])', '[]');
```

Comment:
Surely a user is never going to be asked this as we've disabled submitting/evaluating transactions from the command palette?